File: drimtajm/day1/day1.r
mydata <- read.table("input1.txt")

calculate.fuel <- function(mass) {
  fuel <- (mass %/% 3) - 2
  fuel * (fuel > 0)
}

calculate.total.fuel <- function(mass) {
  total.fuel <- integer(length(mass))
  fuel <- mass
  while (any(fuel > 0)) {
    fuel <- calculate.fuel(fuel)
    total.fuel <- total.fuel + fuel
  }
  print(sum(total.fuel))
}

calculate.total.fuel(mydata)

#> system.time(source("ex1b.r"))
#[1] 4940279
#   user  system elapsed 
#  0.015   0.000   0.016 

File: drimtajm/day1/day1.py
import time

def calcFuel(mass):
    fuel = mass / 3 - 2
    if fuel > 0:
        return fuel
    return 0

def calcTotalFuel():
    with open("input1.txt", "rt") as f:
        total = 0
        for line in f:
            mass = int(line)
            while (mass > 0):
                mass = calcFuel(mass)
                total += mass
    return total

start_t = time.time()
print calcTotalFuel()
stop_t = time.time()
print stop_t - start_t
